### PR TITLE
Don't fail if sysctl doesn't know the max datagram size.

### PR DESCRIPTION
**Pull request by fifteenhex**
…efault instead.

### CODE REVIEW

File: ptftplib/tftpserver.py [hunk 1]
```diff
@@ -70,9 +70,11 @@ def get_ip_config_for_iface(iface):
...
 def get_max_udp_datagram_size():
     """Retrieve the maximum UDP datagram size allowed by the system."""
-    val = subprocess.check_output(['sysctl', '-n', 'net.inet.udp.maxdgram'])
-    return int(val)
-
+    try:
+        val = subprocess.check_output(['sysctl', '-n', 'net.inet.udp.maxdgram'])
+        return int(val)
+    except subprocess.CalledProcessError:
+        return 508
```
Comment (mpetazzoni) [inline ptftplib/tftpserver.py]: Could you define a constant for this "magic" value, and document that constant to explain the choice of this particular value for a fallback UDP datagram size?

**Review by mpetazzoni — CHANGES_REQUESTED** [review]

File: ptftplib/tftpserver.py (hunk 1 above)
Comment (fifteenhex) [reply ptftplib/tftpserver.py]: Thanks for getting back to me. I'll fix this up. :)

Comment (mpetazzoni) [reply ptftplib/tftpserver.py]: @fifteenhex did you get a chance to update this? Would love to merge this PR!

**Comment by rgov** [comment]
I hit this earlier. There is a socket option, `SO_SNDBUF` which you can use to potentially figure out the maximum packet size. But I'm not confident it is correct.

https://stackoverflow.com/questions/66447113/cross-platform-way-to-determine-maximum-udp-datagram-size

Two problems: (1) on Ubuntu 18.04 the value seems ridiculously large, (2) the definition of `SO_SNDBUF` implies you need to do some math to figure out the actual max packet size, which doesn't make much sense to me.

**Comment by alex-eri** [comment]
@rgov SO_SNDBUF not correlates with udp.maxdgram.

I rtsp protocol there is detected by sending big packet and reduce it size while errors on checksum. No way to detect actual safe packet size without interactions with client.

**Comment by rgov** [comment]
He lives! Thanks @mpetazzoni.

**Pull request merged**